drivers/platform/xilinx/xilinx_uart.c

```diff
@@ -324,6 +326,31 @@ static int32_t uart_irq_init(struct no_os_uart_desc *descriptor)
 }
 #endif // XUARTPS_H
 
+#ifdef _XPARAMETERS_PS_H_
+static int32_t irq_setup(struct no_os_irq_ctrl_desc **irq_desc)
+{
+	int32_t status;
+
+	struct xil_irq_init_param p = {
+		.type = IRQ_PS,
+	};
+	struct xil_irq_init_param *platform_irq_init_par = &p;
+	const struct no_os_irq_platform_ops *platform_irq_ops = &xil_irq_ops;
+
+	struct no_os_irq_init_param irq_init_param = {
+		.irq_ctrl_id = XPAR_SCUGIC_SINGLE_DEVICE_ID,
```

Comment:
i also agree it should be but i think it's beyond the scope of this pull-request which just aims to move irq initialization from iio_app to xilinx_uart

Reply:
actually i found a project where this is different, XPAR_INTC_SINGLE_DEVICE_ID instead of XPAR_SCUGIC_SINGLE_DEVICE_ID

https://github.com/analogdevicesinc/no-OS/blob/cbd8405892eeadc4383a6d8ddc3644fb9a6f4743/projects/ad9208/src/parameters.h#L72

Reply:
This code is run only for Xilinx platforms with hard processors (no MB). For these, INTC_DEVICE_ID is set to XPAR_SCUGIC_SINGLE_DEVICE_ID.